app/packages/state/src/hooks/useHover.ts

```diff
@@ -0,0 +1,26 @@
+import { useRef, useState, useEffect } from "react";
+
+function useHover() {
```

Comment:
I would explicitly type what is returned. Right now the type is a an array and not a tuple.